ddtrace/tracer/metrics_test.go

```diff
@@ -71,4 +81,29 @@ func (tg *testGauger) Reset() {
 	tg.mu.Lock()
 	defer tg.mu.Unlock()
 	tg.calls = tg.calls[:0]
+	tg.tags = tg.tags[:0]
+	if tg.waitCh != nil {
+		close(tg.waitCh)
```

Comment:
Do we need `tg.waitCh = nil` here? Otherwise you won't be able to call `Wait` after `Reset`?

Reply:
Good spotting! Pushed a fix.